### PR TITLE
[Doc] Add document for datax and sample codes

### PR DESCRIPTION
**Pull request by morningman**
## Proposed changes

Add documents for datax in `extension` catalog.
Add documents for sampes in `best-practice` catalog.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [ ] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/best-practices/samples.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+└── mini_load
+```
+
+* `connect/`
+
+    该目录下主要展示了各个程序语言连接 Doris 的代码示例。
+    
+* `doris-demo/`
+
+    该目下主要以 Maven 工程的形式，展示了 Doris 多个功能的代码示例。如 spark-connector 和 flink-connector 的使用示例、与 Spring 框架集成的示例、Stream Load 导入示例等等。
```
Comment (EmmyMiao87) [inline docs/zh-CN/best-practices/samples.md:49]: 这四个文件夹感觉并不是平级关系？

File: docs/zh-CN/extending-doris/datax.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+[DataX](https://github.com/alibaba/DataX) doriswriter 插件，用于通过 DataX 同步其他数据源的数据到 Doris 中。
+
+## 编译安装
+
+插件代码位于 [这里](https://github.com/apache/incubator-doris/tree/master/extension/DataX)
+
+该插件通过 Doris 的 Stream Load 功能进行数据的同步和导入。需要配合 DataX 服务使用。
+
+请参阅 [README](https://github.com/apache/incubator-doris/blob/master/extension/DataX/README) 文档进行编译安装操作。
```
Comment (EmmyMiao87) [inline docs/zh-CN/extending-doris/datax.md:37]: What's different between this and [https://github.com/apache/incubator-doris/blob/master/extension/DataX/doriswriter/README]?
Should we keep only one?

File: docs/zh-CN/best-practices/samples.md (hunk 1 above)
Comment (hf200012) [reply docs/zh-CN/best-practices/samples.md:49]: 主要是原先有三个，其实还需要在整理一下按照语言分类，该删除的删除，改迁移到doris-demo下的迁移

File: docs/zh-CN/extending-doris/datax.md (hunk 2 above)
Comment (morningman) [reply docs/zh-CN/extending-doris/datax.md:37]: I will move README of DataX to this document

**Review by EmmyMiao87 — APPROVED** [review]
LGTM

**Pull request merged**